### ISSUE TITLE
XMLSchema.to_dict() leaks elementpath exceptions

### ISSUE THREAD

**Issue by ccc-larc** [sissaschool/xmlschema#136]
Dear all,

When the schema has `xs:date` type with `xs:minInclusive` / `xs:maxInclusive` restriction, using `XMLSchema.to_dict()` with `validation='lax'` leaks exception.

Here is an example:

```python
from xmlschema import XMLSchema


SCHEMA = """
<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema" version="1.0">
    <xs:element name="Date">
        <xs:simpleType>
            <xs:restriction base="xs:date">
                <xs:minInclusive value="2000-01-01"/>
                <xs:maxInclusive value="2099-12-31"/>
            </xs:restriction>
        </xs:simpleType>
    </xs:element>
</xs:schema>
"""


VALID_XML = """<?xml version="1.0" encoding="utf-8"?>
<Date>2019-01-01</Date>
"""


INVALID_XML = """<?xml version="1.0" encoding="utf-8"?>
<Date>2019</Date>
"""


if __name__ == '__main__':
    xs = XMLSchema(SCHEMA)

    data, errors = xs.to_dict(VALID_XML, validation='lax')
    print((data, errors))
    # ('2019-01-01', [])

    data, errors = xs.to_dict(INVALID_XML, validation='lax')
    # elementpath.exceptions.ElementPathTypeError: wrong type <class 'NoneType'> for operand None.
```

### Environment

Python 3.6.8

```
elementpath==1.2.1
xmlschema==1.0.14
```

**Comment by brunato** [comment]
Hi,
found the problem. It was related to `None` value, that is used with _lax_ validation if the value is not decodable to the target type. Now facets are skipped in these cases and also I've inserted the facets validators in a `try-except` statement that intercept possible exceptions and convert them to a validation error. The fix will be available within the next release.
Thank you

**Comment by brunato** [comment]
Fix for this is finally available with release v1.0.15.
Bye
